src/System.Management.Automation/engine/parser/TypeResolver.cs

```diff
@@ -109,6 +117,10 @@ private static Type LookForTypeInAssemblies(TypeName typeName, IEnumerable<Assem
                             }
                         }
                     }
+                    else
+                    {
+                        searchedAssemblies.Add(assembly.FullName);
```

Comment:
The `else` block is when `targetType == null`, because the `if` block above is checking `targetType != null`.

The `alternateNameToFind` is handled before the `if (targetType != null)` check. So once we get to the else block, that means we cannot find target type with or without the alternate name from the current assembly.